code/game/objects/objs.dm

```diff
@@ -93,7 +93,10 @@
 
 
 /obj/attack_ghost(mob/user)
-	ui_interact(user)
+	if(user.client && user.client.inquisitive_ghost)
+		..()
+	else
+		ui_interact(user)

```

Comment:
Just do this, attack_ghost handles the inquisitive check itself.
    ui_interact(user)
    ..()